tools/sof-perf-analyzer.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+
+'''
+The sof_trace_analyzer helps to analyze SOF trace output generated during the running of
+aplay and arecord commands or a test case in sof-test repository. It takes human readable
+trace file (contents from plain mtrace or the decoded dictionary trace) as input, and
+output below analysis results:
+    - Average and maximum MCPS of cpu average and cpu peak per SOF component
+
+Hardcoded value explained:
+DSP_CLK: The DSP clock frequency in Hz.
+DSP_TIMER: The DSP timer clock frequency in Hz.
+'''
+
+import re
+import pathlib
+import argparse
+from typing import Dict
+from typing import List
+from typing import TextIO
+from typing import Optional
+from typing import Generator
+from dataclasses import dataclass
+
+@dataclass()
+class TraceItem:
+    '''The structural representation for a single line of trace'''
+    timestamp: float
+    level: str
+    context: str # the trace context registered with LOG_MODULE_REGISTER
+    func: str
+    msg: str
```

Comment:
I found this `msg` field to be the most confusing part of the design. Maybe it's OK but it needs at least more comments.

This structure represents a parsed one line of logs / messages. So why is a field still called "msg"? Is this the entire line kept for recording purposes?

The purpose of this script is to extract performance numbers. Yet there is no performance number in a parsed line of logs?

See other comment about your 2-steps parsing below.